### ISSUE TITLE
UnboundLocalError bug in callbacks.py

### ISSUE THREAD

**Issue by kieutrang1211** [ivo-1/bomberman_rl#15]
When executing `python3 main.py play --my-agent coli_agent --train 1 --n-rounds 1000 --no-gui` the following error can be obeserved:

```sh
Traceback (most recent call last):
  File "main.py", line 265, in <module>
    main()
  File "main.py", line 253, in main
    world_controller(
  File "main.py", line 73, in world_controller
    world.do_step(user_input)
  File "/home/students/vu/FML/bomberman_rl/environment.py", line 207, in do_step
    self.send_game_events()
  File "/home/students/vu/FML/bomberman_rl/environment.py", line 563, in send_game_events
    a.wait_for_game_event_processing()
  File "/home/students/vu/FML/bomberman_rl/agents.py", line 191, in wait_for_game_event_processing
    self.backend.get("game_events_occurred")
  File "/home/students/vu/FML/bomberman_rl/agents.py", line 328, in get
    return self.get_with_time(expect_name, block, timeout)[0]
  File "/home/students/vu/FML/bomberman_rl/agents.py", line 340, in get_with_time
    raise result
  File "/home/students/vu/FML/bomberman_rl/agents.py", line 296, in process_event
    event_result = getattr(module, event_name)(self.fake_self, *event_args)
  File "/home/students/vu/FML/bomberman_rl/agent_code/coli_agent/train.py", line 91, in game_events_occurred
    new_state = state_to_features(self, new_game_state, self.history)
  File "/home/students/vu/FML/bomberman_rl/agent_code/coli_agent/callbacks.py", line 560, in state_to_features
    f"Shortest path feature says: {_shortest_path_feature(self, game_state)}"
  File "/home/students/vu/FML/bomberman_rl/agent_code/coli_agent/callbacks.py", line 394, in _shortest_path_feature
    best_other_agent,
UnboundLocalError: local variable 'best_other_agent' referenced before assignment
srun: error: node44: task 0: Exited with exit code 1
```

Simple bug, the variable `best_other_agent` will be defined in the for loop but if the loop won't be reached the statement `shortest_paths_to_coins.append()` will reference it before assignment.

**Comment by kieutrang1211** [comment]
[LocalError.txt](https://github.com/ivo-1/bomberman_rl/files/8259100/LocalError.txt)
Here the tail of the log file.

**Comment by ivo-1** [comment]
Fixed in #16
